be/src/agent/task_worker_pool.cpp

```diff
@@ -618,6 +619,116 @@ void* TaskWorkerPool::_push_worker_thread_callback(void* arg_this) {
+                            << ", skip publish version";
+                    continue;
+                }
+            }
+
+            map<TabletInfo, RowsetSharedPtr> tablet_related_rs;
+            StorageEngine::instance()->txn_manager()->get_txn_related_tablets(transaction_id, partition_id,
+                                                                              &tablet_related_rs);
+
+            TVersion version = par_ver_info.version;
+
+            std::vector<TabletInfo> tablet_infos;
+            tablet_infos.reserve(tablet_related_rs.size());
+            std::vector<Status> statuses(tablet_related_rs.size(), Status::OK());
+
+            size_t idx = 0;
+            // each tablet
+            for (auto& tablet_rs : tablet_related_rs) {
+                tablet_infos.push_back(tablet_rs.first);
+
+                threadpool->submit_func([&worker_pool_this, &tablet_rs, &statuses, idx, &version, &transaction_id,
+                                         &partition_id]() {
+                    const TabletInfo& tablet_info = tablet_rs.first;
+                    const RowsetSharedPtr& rowset = tablet_rs.second;
+                    auto& status = statuses[idx];
+                    // if rowset is null, it means this be received write task, but failed during write
+                    // and receive fe's publish version task
+                    // this be must return as an error tablet
+                    if (rowset == nullptr) {
+                        LOG(WARNING) << "Not found rowset of tablet: " << tablet_info.tablet_id << ", txn_id "
+                                     << transaction_id;
+                        status = Status::NotFound(fmt::format("Not found rowset of tablet: {}, txn_id: {}",
+                                                              tablet_info.tablet_id, transaction_id));
+                        return;
+                    }
+                    EnginePublishVersionTask engine_task(transaction_id, partition_id, version, tablet_info, rowset);
+
+                    status = worker_pool_this->_env->storage_engine()->execute_task(&engine_task);
+                });
+
+                ++idx;
+            }
+
+            // Wait until that all jobs in threadpool are done.
+            threadpool->wait();
+
+            for (size_t i = 0; i < tablet_infos.size(); ++i) {
+                const auto& tablet_info = tablet_infos[i];
+                const auto& status = statuses[i];
+                if (!status.ok()) {
+                    error_tablet_ids->push_back(tablet_info.tablet_id);
+                    // Use the first non-ok status as error_status.
+                    if (!error_status.ok()) {
+                        error_status = status;
+                    }
+                    continue;
+                }
+                if (publish_version_req.strict_mode) partition_related_tablet_infos.erase(tablet_info);
```

Comment:
When to set the strict_mode?

Reply:
It seems that the `strict_mode` is not used any more. I've remove the code about the `strict_mode` and added a `Deprecated` flag to the `.thrift`.